ACE/ace/CDR_Base.cpp

```diff
@@ -948,8 +951,8 @@ void ACE_CDR::Fixed::normalize (UShort min_scale)
 
 ACE_CDR::Fixed ACE_CDR::Fixed::from_string (const char *str)
 {
-  const bool negative = str && *str == '-';
-  if (negative || (str && *str == '+'))
+  const bool negative = *str == '-';
```

Comment:
Sounds good – thanks for following thi sup, Adam!

From: Adam Mitz [mailto:notifications@github.com]
Sent: Wednesday, July 15, 2015 6:06 PM
To: DOCGroup/ATCD
Cc: Steve Huston
Subject: Re: [ATCD] Fixed coverity issues 1297401 and 1297404 (#76)


In ACE/ace/CDR_Base.cpp<https://github.com/DOCGroup/ATCD/pull/76#discussion_r34735389>:

> @@ -948,8 +951,8 @@ void ACE_CDR::Fixed::normalize (UShort min_scale)

>

>  ACE_CDR::Fixed ACE_CDR::Fixed::from_string (const char *str)

>  {

> -  const bool negative = str && *str == '-';

> -  if (negative || (str && *str == '+'))

> +  const bool negative = *str == '-';

I went back to coverity's web interface and I couldn't see any outstanding issues in CDR_Base. I'm going to consider this one closed unless I hear otherwise.

—
Reply to this email directly or view it on GitHub<https://github.com/DOCGroup/ATCD/pull/76/files#r34735389>.